lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.spec.ts

```diff
@@ -296,203 +290,166 @@ describe('DateWidgetComponent', () => {
                 });
 
                 fixture.detectChanges();
                 await fixture.whenStable();
 
-                expect(widget.field.maxValue).toEqual('30-07-2022');
-                expect(widget.minDate).toBeUndefined();
-                expect(widget.field.minValue).toBeNull();
-            });
+                const expectedMaxDate = adapter.today();
 
-            it('should maxValue and minValue be null if maxDateRangeValue and minDateRangeValue are null', async () => {
-                spyOn(adapter, 'today').and.returnValue(new Date('2022-07-22'));
+                expect(adapter.compareDate(widget.maxDate, expectedMaxDate)).toEqual(0);
+                expect(widget.field.maxValue).toBe(todayString);
+            });
 
+            it('should max date be null if maxDateRangeValue is null', async () => {
                 widget.field = new FormFieldModel(form, {
                     type: FormFieldTypes.DATE,
                     dynamicDateRangeSelection: true,
-                    maxDateRangeValue: null,
                     minDateRangeValue: null,
-                    maxValue: null,
-                    minValue: null
+                    maxDateRangeValue: null
                 });
 
                 fixture.detectChanges();
                 await fixture.whenStable();
 
-                expect(widget.minDate).toBeUndefined();
-                expect(widget.maxDate).toBeUndefined();
-                expect(widget.field.minValue).toBeNull();
+                expect(widget.maxDate).toBeNull();
                 expect(widget.field.maxValue).toBeNull();
             });
 
-            it('should maxValue and minValue not be null if maxDateRangeVale and minDateRangeValue are not null', async () => {
-                spyOn(adapter, 'today').and.returnValue(new Date('2022-07-22'));
-
+            it('should max date be today minus abs(maxDateRangeValue) if maxDateRangeValue negative', async () => {
                 widget.field = new FormFieldModel(form, {
                     type: FormFieldTypes.DATE,
                     dynamicDateRangeSelection: true,
-                    maxDateRangeValue: 8,
-                    minDateRangeValue: 10,
-                    maxValue: null,
-                    minValue: null
+                    minDateRangeValue: null,
+                    maxDateRangeValue: -2
                 });
 
                 fixture.detectChanges();
                 await fixture.whenStable();
 
-                expect(widget.field.minValue).toEqual('12-07-2022');
-                expect(widget.field.maxValue).toEqual('30-07-2022');
+                const expectedMaxDate = subDays(adapter.today(), 2);
+
+                expect(adapter.compareDate(widget.maxDate, expectedMaxDate)).toEqual(0);
+                expect(widget.field.maxValue).toBe('20-02-2022');
             });
         });
+
+        it('should min date be today plus minDateRangeValue if minDateRangeValue positive', async () => {
```

Comment:
One disadvantage of using property names is that, when we are going to change the property name, it is easy to overlook the test description. So we can end-up with outdated descriptions

Reply:
Renamed not to use property names